### ISSUE TITLE
[DOC]: Guidance on Setting DFP Alert Thresholds and Improving Alert Quality

### ISSUE THREAD

**Issue by hsin-c** [nv-morpheus/Morpheus#740]
### How would you describe the priority of this documentation request

Low (would be nice)

### Describe the future/missing documentation

It would be helpful to have some documentation that provides guidance on what to do after training a DFP model. The documentation could include:
- Checks to ensure that the training has been set up to yield the best possible results
- Insights into how to set thresholds for alerting based on the model's outputs
- Possible post-processing steps to prioritize the alerts more effectively

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

**Comment by hsin-c** [comment]
Some past materials that could be helpful for this documentation:

0. Training checkups
- Are we using enough data for training? For instance, if it takes a month for the baseline behavior to be learned, and we only use a day of data to train, the model may not be able to properly identify anomalies.
- Are we overfitting on the training set? It's important to use a validation set during training to obtain a validation loss. (The validation set here means a hold-out set that is not used for training, but used in each epoch to obtain an unbiased loss measurement. This helps to understand how well the model is generalizing to new data and avoid overfitting. This is different from a test set, which contains malicious behavior, used to evaluate the fidelity of the model after it is trained.)
- Are we using a validation set to collect loss statistics for z-score calculation? By default, dfencoder uses the train dataset to calculate the loss statistics (mean and standard deviation) for z-score, which can result in a lower standard deviation, leading to higher z-scores. To avoid this, using a validation set can be helpful.
Remember that a validation set can be fed into the fit() function in dfencoder, and it will take care of validation automatically. Don't forget to set arguments run_validation and use_val_for_loss_stats to true to enable this functionality.

1. Grouping by and setting thresholds:
- Tying alerts to a permanent (or semi-permanent) identity can reduce the volume of alerts and help with threat investigation.
- The entity used for grouping (e.g. user account, ARN of processes, AWS role, IP) should be stable and exhibit behavior patterns we want to detect deviations from.
- To set a threshold:
   - Collect inference results for the past month (or another suitable timespan that gives low variance).
   - Group by (entity, date) and find the max/mean/sum anomaly score (z score) for each tuple.
   - Plot the distribution of alert volume (tuple count) over threshold to see how volume changes.
   - If there's an ideal volume in mind, choose the threshold that results in that amount.
   - If there isn't a target volume, look for a steep drop in volume at any threshold and use that as the alerting threshold.
- Collecting results from all models and setting a universal threshold is a good starting point:
    - Using early stopping (a feature provided in dfencoder through the patience model argument) ensures that all models share the same convergence criteria. By further scaling the losses with z-scores, the anomaly score output of each model should be on the same scale and comparable.
- If the alerts are dominated by a few models after analysis and we want more coverage for the more stable models (with lower z-loss), we can consider setting individual thresholds for each model.
    - An automatic way to set thresholds for each model is to set a volume budget for each model and follow the steps above to choose the threshold that will produce the desired volume.

2. Improving alert quality and post-processing
- Once the threshold is defined, we can start analyzing and understanding the set of alerts generated. If there are obvious false positives, we can prioritize alerts by redefining the way the anomaly score is calculated.
- Z-score is an unbounded metric, and sometimes taking the mean of the scores can result in extreme values dominating the final anomaly score. We can normalize individual z-scores before averaging them. Here are some ways to normalize them:
   - Set a maximum z-score cap and linearly map the z-score range from [0, max z-score] to [0, 1] (or any range that makes sense).
   - To maintain continuity, we can use a tuned sigmoid function to map the range from [0, infinity] to [0, 1]. The sigmoid function can be tuned so that the middle region with a steep slope matches the desired [0, max z-score] range.
- After normalization, we can take a mean of the features' z-scores or do a weighted sum. We can tune the weights to reflect the importance of each feature for our specific use case.
